### PR TITLE
Add new job to leave comment to new contributor

### PR DESCRIPTION
**Pull request by aldo-expensify**
Hey @roryabraham , I added you as a reviewer as I think you know very well how this is supposed to be implemented :) 

I didn't spot any problem with the code provided in the GH issue besides changing `çount` for `count` 

### Details

Add new job "newContributorReminder" to `preDeploy.yml`. Steps in this job:

1. Checkout repository
2. Find last merged PR to get author
3. Get count of PR's done by this author
4. If count of PR's is exactly 1, leave a comment in the merged PR reminding the contributor about things to expect/responsibilities.

I have these doubts about this:

1. I would add the `internal QA` label, is that correct in this case?
2. In the first step, should we keep `fetch-depth: 0` for the `uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f`? I left it there because removing it would only make sense to me for performance reasons, but I guess that's not too important here. Fetching with a narrower scope could make it miss somethings we need to find? The PR's?
3. What happen if two different first contributors' PRs are merged about the same time. Would this job be able to run between each of these PRs? or probably one of them would miss it. I'm guessing that this would be a very rare case, and if one contributor misses his comment, it's not a big deal, but anyway I'm curious about that case :)

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/176233

### Tests

- Create a PR and merge it with an account that has not done any PR before
- The PR should get the new comment
- Do a PR again with the same account and merge it
- The new PR shouldn't get the new comment

### QA Steps

Reach out to @aldo-expensify to perform QA of this issue

### CODE REVIEW

**Comment by roryabraham** [comment]
> I would add the internal QA label, is that correct in this case?

We don't yet have an internal QA checklist for New Expensify. You'll just have to remember to QA this in the next deploy cycle by submitting a PR with a new GitHub account (or just verifying that it works for the next first-time contributor PR merged).

**Comment by roryabraham** [comment]
> In the first step, should we keep `fetch-depth: 0`

I think it's safe to remove in this case, so we should.

**Comment by roryabraham** [comment]
> What happen if two different first contributors' PRs are merged about the same time.

Two separate jobs would happen in different GHA runners at the same time. Should still work fine.

File: .github/workflows/preDeploy.yml [hunk 1]
```diff
@@ -137,3 +137,40 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ github.token }}
           SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK }}
+  newContributorReminder:
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:140]: Can we add some empty lines and a comment description between these jobs please?

Comment (roryabraham) [reply .github/workflows/preDeploy.yml:140]: Also, let's call the job `newContributorWelcomeMessage`

File: .github/workflows/preDeploy.yml [hunk 2]
```diff
@@ -137,3 +137,40 @@ jobs:
...
+    runs-on: ubuntu-latest
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:141]: For the sake of general efficiency and noise reduction, let's skip this job for `OSBotify`:

```yaml
newContributorWelcomeMessage:
  if: ${{ github.actor != 'OSBotify' }}
  runs-on: ubuntu-latest
```

File: .github/workflows/preDeploy.yml [hunk 3]
```diff
@@ -137,3 +137,40 @@ jobs:
...
+    steps:
+      # Version: 2.3.4
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          fetch-depth: 0
```
Comment (roryabraham) [inline .github/workflows/preDeploy.yml:146]: As you stated in the PR description, I think we can get rid of this:

```suggestion
```

**Review by roryabraham — CHANGES_REQUESTED** [review]

**Comment by aldo-expensify** [comment]
> > I would add the internal QA label, is that correct in this case?
> 
> We don't yet have an internal QA checklist for New Expensify. You'll just have to remember to QA this in the next deploy cycle by submitting a PR with a new GitHub account (or just verifying that it works for the next first-time contributor PR merged).

Sounds good. [NO QA] should stay there, right? because this won't be done by applause

File: .github/workflows/preDeploy.yml (hunk 2 above)
Comment (aldo-expensify) [reply .github/workflows/preDeploy.yml:141]: Ok! Small question, I see in the documentation that the `${{ }}` are not needed in `jobs.if` ([doc](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif)). Should I still add it? I can see that in other cases in this file there are `${{ }}`, but in other files there isn't.

**Comment by aldo-expensify** [comment]
Update: addressed comments!

**Comment by roryabraham** [comment]
> Sounds good. [NO QA] should stay there, right? because this won't be done by applause

Yep, or you can utilize applause and make the QA steps "reach out to @aldo-expensify to perform QA of this issue".

File: .github/workflows/preDeploy.yml (hunk 2 above)
Comment (roryabraham) [reply .github/workflows/preDeploy.yml:142]: I guess it's up to you, but I recommend just including it for consistency. That way we don't have to remember when `${{ }}` is required and when it's not.

**Review by roryabraham — APPROVED** [review]
LGTM

**Review by nickmurray47 — APPROVED** [review]

**Comment by roryabraham** [comment]
Looks like this worked 🎉  https://github.com/Expensify/App/pull/5060#issuecomment-919438004

**Comment by mvtglobally** [comment]
@aldo-expensify, PR has been deployed. Let us know if you want us to perform any QA steps

**Comment by aldo-expensify** [comment]
> @aldo-expensify, PR has been deployed. Let us know if you want us to perform any QA steps

Sorry, I missed this message.

No need @mvtglobally , thanks!

**Pull request merged**